pyatv/mrp/protocol.py

```diff
@@ -133,11 +149,9 @@ def stop(self):
         self._initial_message_sent = False
         self._outstanding = {}
         self.connection.close()
+        self._state = ProtocolState.NotConnected
 
-    async def _connect_and_encrypt(self):
-        if not self.connection.connected:
-            await self.start()
-
+    async def _enable_encryption(self):
         # Encryption can be enabled whenever credentials are available but only
         # after DEVICE_INFORMATION has been sent
         if self.service.credentials and self._initial_message_sent:
```

Comment:
This is just a clean-up I saw now. Since we only call this method after sending the initial message, there's really no point of having a check for that (it will always be `True` here). So you can remove all instances of `self._initial_message_sent` in the file.